src/main/java/com/sihenzhang/crockpot/mixin/PiglinTasksMixin.java

```diff
@@ -137,19 +136,19 @@ private static void pickUpItemHandler(PiglinEntity piglinEntity, ItemEntity item
     @Inject(
             method = "stopHoldingOffHandItem(Lnet/minecraft/entity/monster/piglin/PiglinEntity;Z)V",
             at = @At(
-                    value = "JUMP",
-                    ordinal = 1,
-                    opcode = 154
+                    value = "INVOKE",
+                    target="Lnet/minecraft/entity/monster/piglin/PiglinEntity;getMainHandItem()Lnet/minecraft/item/ItemStack;"
             ),
             cancellable = true,
             locals = LocalCapture.CAPTURE_FAILHARD
     )
-    private static void stopHoldingOffHandItemHandler(PiglinEntity piglinEntity, boolean isNotHurt, CallbackInfo ci, ItemStack offhandStack, boolean isPiglinCurrency, boolean canBeEquipped) {
+    private static void stopHoldingOffHandItemHandler(PiglinEntity piglinEntity, boolean isNotHurt, CallbackInfo ci, ItemStack offhandStack, boolean isPiglinCurrency) {
         PiglinBarteringRecipe recipe;
-        if (isNotHurt && !canBeEquipped && !offhandStack.getItem().is(ItemTags.PIGLIN_REPELLENTS) && !isFood(offhandStack.getItem()) && (recipe = PiglinBarteringRecipe.getRecipeFor(offhandStack, piglinEntity.level.getRecipeManager())) != null) {
```

Comment:
In our opinion, vanilla logic should take precedence over everything else. So if an item can be equipped, it will be equipped first.

Reply:
Sure, But this already injuct in an if of canBeEquiped

Reply:
Minecraft code:
```
if(!canBeEquiped){
  entity.getMainHandItem() 
}
```
so adding one more is not necessary